sw/device/tests/entropy_src_csrng_test.c

```diff
@@ -248,16 +248,26 @@ static void test_edn_cmd_done(const dif_edn_seed_material_t *seed_material) {
 
   plic_interrupts_enable();
 
-  // Generate enough entropy for the otbn randomness test.
-  // The len provided here is in number of words as opposed to num of 128b
-  // blocks. The requested len **must** also be equal to the amount of entroy
-  // required by the OTBN randomness test, otherwise the Generate command will
+  // The EDN0 is connected to other peripherals that regularly request entropy
+  // so we keep generating entropy on the EDN0 to make sure that the OTBN
+  // gets enough to finish the test. The EDN1 is only connected to the OTBN
+  // so we generate exactly the amount of entropy necessary for the test and not
+  // more otherwise the Generate command will
   // not be fully executed, causing a hang in the `irq_block_wait()` calls
-  // following the end of the OTBN test.
-  // EDN0: 20 words = 5x128b blocks
-  // EDN1: 44 words = 11x128b blocks
+  // following the end of the OTBN test. The OTBN test reads `RND` 5 times
+  // and each read consumes 256b of entropy. The EDN1 generates entropy per
+  // blocks of 128b so we need to generate 10 blocks.
+  const size_t kEdnBlockSizeBits = 128;     // Each EDN block contains 128b.
+  const size_t kOtbnRequestSizeBits = 256;  // Each OTBN request requires 256b.
```

Comment:
As far as I can see these are always converted to words - maybe it would be better to define them like that?

```suggestion
  const size_t kEdnBlockSizeWords = 4;     // Each EDN block contains 128b.
  const size_t kOtbnRequestSizeWords = 8;  // Each OTBN request requires 256b.
```

Reply:
I would rather avoid this because the OTBN code can request either words or side register and each requests uses 256b independently of the request (ie request a word of entropy will consume 256b). So it's really the number of requests that counts.

Reply:
Hmm okay, fair enough